src/Actions/RunModelShowCommand.php

```diff
@@ -2,31 +2,113 @@
     {
         $relationships = implode(',', Arr::flatten(config('modeltyper.custom_relationships', [])));
-        $exitCode = Artisan::call("model:typer-show {$model} --json --no-interaction --custom-relationships=$relationships");
 
-        if ($exitCode !== 0) {
-            throw new Exception('You may need to install the doctrine/dbal package to use this command.');
+        $commandArgs = [
+            'model' => $model,
+            '--json' => true,
+            '--no-interaction' => true,
+            '--throw-exceptions' => true,
+        ];
+
+        if ($resolveAbstract) {
+            $commandArgs['--resolve-abstract'] = true;
+        }
+
+        if (! empty($relationships)) {
+            $commandArgs['--custom-relationships'] = $relationships;
+        }
+
+        $command = ShowModelCommand::class;
+
+        try {
+            $this->runCommandWithoutMockOutput($command, $commandArgs);
+        } catch(CommandException $exception) {
+            $msg = "Command '$command' failed:" . PHP_EOL . $exception->getMessage();
+            throw new NestedCommandException($msg, Command::FAILURE, $exception);
+        }
+
+        $output = Artisan::output();
+
+        // NOTE this check should not fail under normal circumstances, but might be useful to catch
+        // unexpected errors if running the command without mock fails for some reason.
+        if (empty($output)) {
+            $msg = "Could not resolve types for model '$model', Artisan::output() is empty.";
+            $msg .= PHP_EOL . 'If you are running tests, make sure to set {public $mockConsoleOutput = false;}';
+            throw new NestedCommandException($msg);
+        }
+
+        return json_decode($output, true);
+    }
+
+    /**
+     * Run an Artisan command without testing mock output to prevent the command output
+     * from being intercepted by mock output handler. Resets the mock output handler after
+     * command execution so that mock testing is not disturbed.
+     *
+     * @param  string  $command Name of the command to run.
+     * @param  array<string, mixed>  $args Arguments for the command.
+     * @return int $exitCode Exit code returned by the command.
+     */
+    private function runCommandWithoutMockOutput(string $command, array $args = []): int
+    {
+        $originalOutput = $this->outputMocked() ? $this->app->get(OutputStyle::class) : false;
+        $this->withoutMockingConsoleOutput();
+
+        $exitCode = Artisan::call($command, $args);
+
+        if ($originalOutput) {
+            $this->app->bind(OutputStyle::class, fn () => $originalOutput);
+        }
+
+        return $exitCode;
+    }
+
+    /**
+     * Check if the console output is being mocked.
+     *
+     * @return bool $mocked
+     */
+    private function outputMocked(): bool
+    {
+        if (! app()->runningInConsole()) {
+            return false;
+        }
+
+        if (! config('app.env') === 'testing') {
```

Comment:
maybe you can replace with
```php
if (! app()->environment('testing')) {
 // ...
}
```

Reply:
Resolved by [18525a3](https://github.com/fumeapp/modeltyper/pull/42/commits/18525a3ead34a9fe1b845b7afaf96993398790dc)